### PR TITLE
Update build tools

### PR DESCRIPTION
**Pull request by thiamsantos**
*Sorry if any of the changes are out of scope*.

**CHANGELOG**
- Use [rollup](http://rollupjs.org/) instead of [webpack](https://webpack.js.org/) #225.
- Remove unnecessary file `eslintignore`.
- Move [babel](https://babeljs.io/) and [eslint](http://eslint.org/) configurations to `package.json`.
- Change scripts that use environment variables [because windows doesn't work well with them](https://github.com/evgenyrodionov/redux-logger/issues/225#issuecomment-295446452).
- Update [`deep-diff`](https://www.npmjs.com/package/deep-diff).
- Remove [`open-url`](https://www.npmjs.com/package/open-url) because [`http-server`](https://www.npmjs.com/package/http-server) already has an [option to open the browser after starting the server](https://www.npmjs.com/package/http-server#available-options).
- Remove `babel-cli` and others unused plugins and presets.
- Expose files using ES2015 modules through the properties [`pkg.module` and `pkg.jsnext:main`](https://github.com/rollup/rollup/wiki/pkg.module).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,23 +2,35 @@
...
-    "build": "npm run build:lib && npm run build:umd && npm run build:umd:min",
-    "precommit": "npm run test",
-    "prepublish": "npm run clean && npm run test:production && npm run build"
+    "clean": "rimraf dist",
+    "build": "npm run rollup && npm run uglify",
+    "precommit": "npm test",
+    "prepublish": "npm run clean && npm test && npm run build",
+    "uglify": "uglifyjs dist/index.umd.js -cm -o dist/index.umd.min.js",
+    "rollup": "node rollup.config.js"
+  },
```
Comment (evgenyrodionov) [inline package.json]: Can move scripts as they appears? For example, `build` run `rollup`, so `rollup` must be higher than `build`

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+const rollup = require('rollup').rollup;
```
Comment (evgenyrodionov) [inline rollup.config.js]: We don't need non-minified versions, so you can produce minified versions right here

Comment (thiamsantos) [inline rollup.config.js]: Sure. But I'm still in doubt if I should bundle two different version, one using UMD and another CommonJS modules. Maybe just one is enough. What do you think @evgenyrodionov ?

Comment (evgenyrodionov) [inline rollup.config.js]: I think UMD is enough

**Comment by thiamsantos** [comment]
Abu thoughts @evgenyrodionov?

**Comment by evgenyrodionov** [comment]
Looks good, but I need test it. On Monday, I guess.

**Comment by thiamsantos** [comment]
I hope you don't mind @evgenyrodionov. I am taking advantage of this pull request to add some tests.

**Comment by evgenyrodionov** [comment]
Thank you! It looks pretty awesome.

**Comment by thiamsantos** [comment]
Thanks @evgenyrodionov!!!

**Comment by thiamsantos** [comment]
> Expose files using ES2015 modules through the properties pkg.module and pkg.jsnext:main.

As I mentioned above if you use a module bundler like webpack or rollup, it will get the code from the source files. That can cause problems like #229 when you don't use the babel plugin to handle the object spread operator. So I think that we can either explain that on readme and ask for people use the plugin or just expose the bundled file as was made before and prevent these problems. What do you think @evgenyrodionov?

**Comment by evgenyrodionov** [comment]
Object spread operator already fixed in 3.0.3, it's stage 3 proposal (candidate), so it was my mistake.

**Pull request merged**